plugins/sourcegraph/access_token.go

```diff
@@ -0,0 +1,64 @@
+package sourcegraph
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func AccessToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.AccessToken,
+		DocsURL: sdk.URL("https://docs.sourcegraph.com/cli"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Username,
+				MarkdownDescription: "Username on your Sourcegraph instance.",
+				Secret:              false,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+						Specific:  []rune{'.', '-', '_'},
+					},
+				},
+				Optional: true,
+			},
+			{
+				Name:                fieldname.HostAddress,
+				MarkdownDescription: "Base URL for your Sourcegraph instance.",
+				Secret:              false,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to Sourcegraph.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 40,
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"SRC_ENDPOINT":     fieldname.HostAddress,
+	"SRC_ACCESS_TOKEN": fieldname.Token,
```

Comment:
We've been using `Token` as the field name elsewhere, as a shorthand to avoid too much duplication with the credential type name.

Reply:
Okay, in that case this shouldn't be a blocker for this PR, and we can re-assess for all plugins together as an independent effort in case we want to change this.

Reply:
i.e. we can leave as is: `Token` for now.